.github/workflows/release.yaml

```diff
@@ -1,280 +1,358 @@
+          mkdir -p ${{ env.OUT_DIR }}
+          wasm-bindgen target/${{ env.TARGET }}/${{ env.PROFILE }}/${{ env.EXE_NAME }}.wasm --out-dir ${{ env.OUT_DIR }} --out-name ${{ env.EXE_NAME }} --no-typescript --target web 
+          cp -r assets web/* ${{ env.OUT_DIR }} || true # Ignore error if `assets` or `web` folder does not exist
+
+      - name: Optimize Wasm
+        uses: NiklasEi/wasm-opt-action@v2
+        with:
+          file: ${{ env.OUT_DIR }}/*.wasm
 
-      - name: Package as a zip
-        working-directory: ./wasm
+      - name: Compress package
         run: |
-          zip --recurse-paths ../${{ env.binary }}.zip .
+          zip --recurse-paths ${{ env.PACKAGE_NAME }}-${{ env.PLATFORM }}.zip ${{ env.OUT_DIR }}
 
-      - name: Upload binaries to artifacts
+      - name: Upload package to artifacts
         uses: actions/upload-artifact@v3
         with:
-          path: ${{ env.binary }}.zip
-          name: wasm
+          path: ${{ env.PACKAGE_NAME }}-${{ env.PLATFORM }}.zip
+          name: ${{ env.PLATFORM }}
           retention-days: 1
 
-      - name: Upload binaries to release
-        if: ${{ env.add_binaries_to_github_release == 'true' }}
+      - name: Upload package to Github release
+        if: ${{ env.UPLOAD_PACKAGES_TO_GITHUB_RELEASE == 'true' }}
         uses: svenstaro/upload-release-action@v2
         with:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: ${{ env.binary }}.zip
-          asset_name: ${{ env.binary }}-wasm-${{ steps.get_version.outputs.tag }}.zip
+          file: ${{ env.PACKAGE_NAME }}-${{ env.PLATFORM }}.zip
+          asset_name: ${{ env.PACKAGE_NAME }}-${{ env.VERSION }}-${{ env.PLATFORM }}.zip
+          release_name: ${{ env.VERSION }}
           tag: ${{ github.ref }}
           overwrite: true
 
-  # Build for Linux
-  release-linux:
+  # Build for Linux.
+  build-for-linux:
     runs-on: ubuntu-latest
+    needs: get-version
+    env:
+      TARGET: x86_64-unknown-linux-gnu
+      PROFILE: release
+      PLATFORM: linux
+      OUT_DIR: build/${{ env.PLATFORM }}
+      VERSION: ${{ needs.get-version.outputs.version }}
 
     steps:
-      - uses: olegtarasov/get-tag@v2.1.2
-        id: get_version
-      - uses: actions/checkout@v4
+      - name: Checkout repository
+        uses: actions/checkout@v4
         with:
-          lfs: ${{ env.use_git_lfs }}
-      - uses: dtolnay/rust-toolchain@stable
+          lfs: ${{ env.USE_GIT_LFS }}
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
         with:
-          targets: x86_64-unknown-linux-gnu
-      - name: install dependencies
+          targets: ${{ env.TARGET }}
+
+      - name: Install dependencies
         run: |
           sudo apt-get update; sudo apt-get install pkg-config libx11-dev libasound2-dev libudev-dev libxcb-render0-dev libxcb-shape0-dev libxcb-xfixes0-dev
```

Comment:
A few of these dependencies can be removed; I believe you only need the 4 listed [here](https://github.com/bevyengine/bevy/blob/09d86bfb96ccb66020c38485647c002dcfa37956/.github/actions/install-linux-deps/action.yml#L45).

Reply:
```suggestion
          sudo apt-get update; sudo apt-get install libasound2-dev libudev-dev libwayland-dev libxkbcommon-dev
```